blaze/compute/mongo.py

```diff
@@ -347,6 +352,11 @@ def compute_up(t, q, **kwargs):
                   Second: 'second'}
 
 
+@dispatch(Field, Database)
+def compute_up(expr, data, **kwargs):
+    return MongoQuery(getattr(data, expr._name), [])
```

Comment:
Ideally you would only need to do `return getattr(data, expr._name)`.  Pre_compute should trigger every time the type of the data changes.